src/Lecture2.hs

```diff
@@ -62,7 +68,11 @@ lazyProduct = error "TODO"
 "ccaabb"
 -}
 duplicate :: [a] -> [a]
-duplicate = error "TODO"
+duplicate list = makeDuplicateList [] list
+          where
+            makeDuplicateList :: [a] -> [a] -> [a]
+            makeDuplicateList l [] = l
+            makeDuplicateList l (x:xs) = makeDuplicateList (l ++ [x,x])  xs
```

Comment:
It is more efficient to construct lists using the cons operator (`:`) directly without introducing intermediate values or adding lists with `(++)` whenever possible. Adding elements to the end of the list is rather slow 🐌

That's why current implementation also hangs on infinite lists.